tests/e2e/specs/in-game/one-offs/5_fives.spec.js

```diff
@@ -367,15 +370,34 @@ describe('FIVES', () => {
           deck: [],
         });
         cy.get('#deck').click();
+        cy.get('#deck').should('contain', '(1)');
+
         cy.drawCardOpponent();
 
         cy.get('#deck').should('contain', 'PASS');
         cy.get('[data-player-hand-card=5-3]').click();
         cy.get('[data-move-choice=oneOff]').should('have.class', 'v-card--disabled');
+
+        // Forcibly make request to play 5 and confirm response is error
+        cy.window()
+          .its('cuttle.gameStore')
+          .then(async (gameStore) => {
+            const fiveId = gameStore.player.hand.find((card) => card.rank === 5 && card.suit === 3).id;
+
+            try {
+              const res = await gameStore.requestPlayOneOff(fiveId);
+              // If the promise resolves, this assertion will fail the test (as expected in your case)
+              expect(true).to.eq(
+                false,
+                `Expected request to resolve five without discarding to error, but instead came back 200: ${res}`,
+              );
+            } catch (err) {
+              // If the promise rejects, this will handle the error
+              expect(err).to.eq('game.snackbar.oneOffs.emptyDeck');
+            }
+          });
```

Comment:
This took some fancy footwork to get cypress to play nice with the promise returned by the pinia store action `gameStore.requestResolveFive()`. We had a way of doing it originally that looked correct and would show the result of the assertion as passing, but it didn't actually fail the test when the assertion failed, due to a race condition.

I think there is an issue where `cy.window().its().then()` doesn't actually create a promise in such a way that the inner callback would bubble up. By using `async/await` here we can ensure that cypress will actually wait for the result of the async call and fail the test if the request comes back 200 or if the server error doesn't match the expected string.

Reply:
I absolutely love stuff like this and it’s why I still have a problem with Cypress using a “promise-like” format. Still, awesome find!